packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -1067,11 +1067,26 @@ class _CupertinoTextFieldState extends State<CupertinoTextField> with Restoratio
     );
   }
 
+  KeyEventResult _handleRawKeyEvent(FocusNode node, RawKeyEvent event) {
+    assert(node.hasFocus);
+    // TextField use the `enter` to finish the input or create a new line, and the
+    // `space` is a very common input character, so we default to terminal those
+    // two keys, otherwise, when its ancestor handles the two keys(such as `ListTile`),
+    // the functions of `TextField` will be abnormal.
+    if (event.logicalKey == LogicalKeyboardKey.space
+        || event.logicalKey == LogicalKeyboardKey.enter) {
+      return KeyEventResult.skipRemainingHandlers;
+    }
+    return KeyEventResult.ignored;
+  }
+
   @override
   Widget build(BuildContext context) {
     super.build(context); // See AutomaticKeepAliveClientMixin.
     assert(debugCheckHasDirectionality(context));
     final TextEditingController controller = _effectiveController;
+    final FocusNode focusNode = _effectiveFocusNode;
+    focusNode.onKey ??= _handleRawKeyEvent;
```

Comment:
This worries me.
If `_effectiveFocusNode` came from `widget.focusNode`, it will change its key handler. Additionally, if `focusNode` doesn't have `onKey`, then the developer must remember to add your logic back. 

I think we might as well add another `Focus` for your additional logic, so that it stays on.

Reply:
You are so right!